analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -110,14 +108,11 @@ public abstract class UtilityAnalyzerBase<TSyntaxKind, TMessage> : UtilityAnalyz
                         .Concat(treeMessages)
                         .WhereNotNull()
                         .ToArray();
-                    lock (FileWriteLock)
```

Comment:
This one is not needed, even different TFMs will have different directories to write to.